.github/workflows/camunda-ex-versioned-java-client.yaml.yml

```diff
@@ -0,0 +1,218 @@
+          fi
+
+          # Convert to JSON arrays
+          CLIENT_JSON=$(echo "$CLIENT_VERSIONS" | jq -R -s -c 'split("\n") | map(select(length > 0))')
+          SERVER_JSON=$(echo "$SERVER_VERSIONS" | jq -R -s -c 'split("\n") | map(select(length > 0))')
+
+          echo "client-versions=$CLIENT_JSON" >> "$GITHUB_OUTPUT"
+          echo "server-versions=$SERVER_JSON" >> "$GITHUB_OUTPUT"
+
+          echo "Found client versions: $CLIENT_JSON"
+          echo "Found server versions: $SERVER_JSON"
+
+      - name: Create test matrix
+        id: create-matrix
+        shell: bash
+        run: |
+          set -e
+
+          CLIENT_VERSIONS='${{ steps.get-versions.outputs.client-versions }}'
+          SERVER_VERSIONS='${{ steps.get-versions.outputs.server-versions }}'
+
+          # Create matrix: each client version tested against ALL server versions
+          MATRIX=$(jq -n \
+            --argjson clients "$CLIENT_VERSIONS" \
+            --argjson servers "$SERVER_VERSIONS" \
+            '{include: [
+              $clients[] as $client |
+              $servers[] as $server |
+              {
+                "client-version": $client,
+                "server-version": $server,
+                "scenario": ($client + "-vs-" + $server)
+              }
+            ]}')
+
+          echo "matrix=$MATRIX" >> "$GITHUB_OUTPUT"
+
+          echo "Generated test matrix:"
+          echo "$MATRIX" | jq .
+
+          # Calculate and log matrix size
+          MATRIX_SIZE=$(echo "$MATRIX" | jq '.include | length')
+          echo "Total test combinations: $MATRIX_SIZE"
+
+  test-compatibility:
+    name: "${{ matrix.client-version }} → ${{ matrix.server-version }}"
+    needs: discover-versions
+    if: needs.discover-versions.outputs.matrix != ''
+    timeout-minutes: 30
+    permissions: {}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 10
+      matrix: ${{ fromJson(needs.discover-versions.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ matrix.client-version }}
+          fetch-depth: 0
+
+      - uses: ./.github/actions/setup-build
+        with:
+          vault-address: ${{ secrets.VAULT_ADDR }}
+          vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
+          vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
+
+      - name: Create build output log file
+        run: echo "BUILD_OUTPUT_FILE_PATH=$(mktemp)" >> "$GITHUB_ENV"
+
+      # Run versioned tests with specified client and server versions
+      - name: Test client ${{ matrix.client-version }} against server ${{ matrix.server-version }}
+        shell: bash
+        run: >
+          ./mvnw -B --no-snapshot-updates
+          -D maven.javadoc.skip=true
+          -D skipUTs -D skipChecks
+          -D failsafe.rerunFailingTestsCount=3
+          -D camunda.test.version=${{ matrix.server-version }}
+          -D camunda.client.version=${{ matrix.client-version }}
```

Comment:
💭  just to confirm my understanding, the client version becomes effective as this affects the client dependency of the QA module, the server version is picked up by the `VersionedTestExtension` which then runs the specified version via docker, right?